src/dash-playlist-loader.js

```diff
@@ -134,48 +250,191 @@ export default class DashPlaylistLoader extends EventTarget {
+      }
+
+      this.syncClientServerClock_(this.onClientServerClockSync_.bind(this));
+    });
+  }
+
+  /**
+   * Parses the master xml for UTCTiming node to sync the client clock to the server
+   * clock. If the UTCTiming node requires a HEAD or GET request, that request is made.
+   *
+   * @param {Function} done
+   *        Function to call when clock sync has completed
+   */
+  syncClientServerClock_(done) {
+    const utcTiming = mpdParser.parseUTCTiming(this.masterXml_);
+
+    // No UTCTiming element found in the mpd. Use Date header from mpd request as the
+    // server clock
+    if (utcTiming === null) {
+      this.clientOffset_ = this.masterLoaded_ - Date.now();
+      return done();
+    }
+
+    if (utcTiming.method === 'DIRECT') {
+      this.clientOffset_ = utcTiming.value - Date.now();
+      return done();
+    }
+
+    this.request = this.hls_.xhr({
+      uri: resolveUrl(this.srcUrl, utcTiming.value),
+      method: utcTiming.method,
+      withCredentials: this.withCredentials
+    }, (error, req) => {
+      // disposed
+      if (!this.request) {
+        return;
+      }
+
+      if (error) {
+        // sync request failed, fall back to using date header from mpd
+        // TODO: log warning
+        this.clientOffset_ = this.masterLoaded_ - Date.now();
+        return done();
+      }
+
+      let serverTime;
+
+      if (utcTiming.method === 'HEAD') {
+        if (!req.responseHeaders || !req.responseHeaders.date) {
+          // expected date header not preset, fall back to using date header from mpd
+          // TODO: log warning
+          serverTime = this.masterLoaded_;
+        } else {
+          serverTime = Date.parse(req.responseHeaders.date);
         }
+      } else {
+        serverTime = Date.parse(req.responseText);
       }
 
-      // set up phony URIs for the playlists since we won't have external URIs for DASH
-      // but reference playlists by their URI throughout the project
-      for (let i = 0; i < this.master.playlists.length; i++) {
-        const phonyUri = `placeholder-uri-${i}`;
+      this.clientOffset_ = serverTime - Date.now();
+
+      done();
+    });
+  }
+
+  /**
+   * Handler for after client/server clock synchronization has happened. Sets up
+   * xml refresh timer if specificed by the manifest.
+   */
+  onClientServerClockSync_() {
+    this.master = this.parseMasterXml();
+
+    this.state = 'HAVE_MASTER';
 
-        this.master.playlists[i].uri = phonyUri;
-        // set up by URI references
-        this.master.playlists[phonyUri] = this.master.playlists[i];
+    this.trigger('loadedplaylist');
+
+    if (!this.media_) {
+      // no media playlist was specifically selected so start
+      // from the first listed one
+      this.media(this.master.playlists[0]);
+    }
+    // trigger loadedmetadata to resolve setup of media groups
+    // trigger async to mimic behavior of HLS, where it must request a playlist
+    setTimeout(() => {
```

Comment:
It might be best to do explicitly `window.setTimeout` for all occurrences of `setTimeout`